execution_engine/src/storage/global_state/scratch.rs

```
@@ -192,7 +206,12 @@ impl CommitProvider for ScratchGlobalState {
         effects: AdditiveMap<Key, Transform>,
     ) -> Result<Digest, Self::Error> {
         for (key, transform) in effects.into_iter() {
-            let cached_value = self.cache.read().unwrap().get(&key).cloned();
+            let cached_value = self
+                .cache
+                .read()
+                .expect("Cache lock should not be poisoned")
```

We can return an error here instead.